query04.sql

```diff
@@ -0,0 +1,24 @@
+-- 4. Using the _shapes.txt_ file from GTFS bus feed, find the **two** routes with the longest trips. 
+-- In the final query, give the `trip_headsign` that corresponds to the `shape_id` of this route 
+-- and the length of the trip.
+
+--   **Structure:**
+--   ```sql
+--   (
+--       trip_headsign text,  -- Headsign of the trip
+--       trip_length double precision  -- Length of the trip in meters
+--   )
+--   ```
+
+with busRouteLines as (
+	select shape_id, st_transform(st_makeline(the_geom),32129) as the_line_geom
+	from septa_bus_shapes
+	group by shape_id
+)
+
+select distinct(t.trip_headsign) as trip_headsign, st_length(r.the_line_geom) as trip_length
```

Comment:
Just to be clear, `distinct` is not a function. Rather `select distinct` is a type of select. So the `distinct` applies to the entire set of fields you're selecting. This is the same as:
```sql
select distinct
    t.trip_headsign as trip_headsign,
    st_length(r.the_line_geom) as trip_length
from ...
```
The only reason I'm calling it out is because you're writing `distinct` as if it only applies to the `trip_headsign` field and I just want to be clear it does not.